src/spanned.rs

```diff
@@ -120,17 +132,18 @@ where
         self.to_tokens(&mut tokens);
         let mut iter = tokens.into_iter();
         let mut span = match iter.next() {
-            Some(tt) => tt.span(),
+            Some(tt) => check_invalid_span(tt.span()),
             None => {
                 return Span::call_site();
             }
         };
         for tt in iter {
-            if let Some(joined) = span.join(tt.span()) {
+            if let Some(joined) = span.join(check_invalid_span(tt.span())) {
                 span = joined;
             }
         }
-        span
+
+        check_invalid_span(span) // Don't know if this is needed?
```

Comment:
I don't know either. It depends whether joining two valid spans can produce an invalid span. Possibly, in the case that the input spans are not contiguous? I am fine with keeping this line.